### PR TITLE
Uncertainty in heatmaps

### PR DESCRIPTION
**Pull request by Priscilla-AL**
# <Feature- Uncertainty on heatmaps> 

**_WHAT DOES THIS PR DO?_**
Add uncertainty to predictions (prediction interval)

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
https://airqoteam.atlassian.net/jira/software/projects/PLAT/boards/41?assignee=5d91eed256bb950dd6489270&selectedIssue=PLAT-553

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/Uncertainty-in-heatmaps

**_HOW DO I TEST OUT THIS PR?_**
cd AirQo-api/src/predict
python -m venv [DESTINATION_FOLDER]
source [DESTINATION_FOLDER]/Scripts/activate
pip install -r requirements.txt
flask run

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
http://127.0.0.1:5000/api/v1/predict/heatmap

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
NA
**_ARE THERE ANY RELATED PRs?_**
NA

### CODE REVIEW

**Comment by ja-odur** [comment]
@Priscilla-AL I am also now getting the issue on `macOS`. I will look into it.

**Comment by Priscilla-AL** [comment]
> @Priscilla-AL I am also now getting the issue on `macOS`. I will look into it.

Okay, Joseph. Thank you.

File: src/predict/gcp_helpers/cloud_function.py [hunk 1]
```diff
@@ -122,15 +123,22 @@ def predict_model(m):
     locations_flat = np.c_[locations[0].flatten(),locations[1].flatten()]
     pred_set = np.c_[locations_flat,np.full(locations_flat.shape[0],time)]
     mean, var = m.predict_f(pred_set)
-
+    
     means = mean.numpy().flatten()
     variances = var.numpy().flatten()
+    std_dev = np.sqrt(variances)
+    # calculate prediction interval
+    interval = 1.96 * std_dev
```
Comment (danielogen) [inline src/predict/gcp_helpers/cloud_function.py:131]: Hi @Priscilla-AL is `1.96` always a constant value or expected to change sometime in the future?

Comment (danielogen) [reply src/predict/gcp_helpers/cloud_function.py:131]: okay...no worries, I see its still work in progress

**Review by ja-odur — COMMENTED** [review]
I am still failing to get the interval in the response. Even after running the cloud function using `python cloud_function.py`. Is there something that I am missing.

**Comment by Priscilla-AL** [comment]
> I am still failing to get the interval in the response. Even after running the cloud function using `python cloud_function.py`. Is there something that I am missing.

Do you have the GCP key?

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by Baalmart — APPROVED** [review]
This works as expected but many things are missing in the PR description, I had to reach out for assistance. 

- Could we have these running instructions somewhere in a README? I had to use the following command to install a version of python on my MAC.
`CFLAGS="-I$(brew --prefix openssl)/include -I$(brew --prefix bzip2)/include -I$(brew --prefix readline)/include -I$(xcrun --show-sdk-path)/usr/include" LDFLAGS="-L$(brew --prefix openssl)/lib -L$(brew --prefix readline)/lib -L$(brew --prefix zlib)/lib -L$(brew --prefix bzip2)/lib" pyenv install --patch 3.6.5 < <(curl -sSL https://github.com/python/cpython/commit/8ea6353.patch\?full_index\=1)`
- How are we separating environments?
- Could we also have this model handled independently in our mono repo for easy visibility/maintainability using our workflow?

**Pull request merged**